students/douglas_klos/session3/mailroom.py

```diff
@@ -102,17 +72,39 @@ def add_donor(name_input):
             name_input = ''
         if name_input in ('q', 'Q'):
             return
-    donations = get_new_donations()
 
-    mailroom_db.extend([(name_input, donations)])
+    mailroom_db.extend([(name_input, [])])
 
 
-def display_database():
-    """ Displays all donors and their donations """
+def add_donation():
+    """ Add new donations to a current donor """
 
-    print()
-    for donor, donations in mailroom_db:
-        print(f'{donor:>24} : {donations:}')
+    donations = []
+    display_database()
+    while True:
+        name = input('\nPlease select a donor to add a donation to: ')
+        if name.lower() in ('q', 'quit'):
+            return
+        elif not any(name in sub_list for sub_list in mailroom_db):
+            print('Donor not found, please select from list')
+        else:
+            break
+
+    while True:
+        try:
+            donation_input = input('Please enter a donation: ')
```

Comment:
cast as a float here and you don't have to do it every time you use it.

Reply:
Thank you for the suggestion, however I was looking for feedback on session4 mailroom2 (which has the dictionary changes) and trigrams. That said I was able to make the float cast change in the session4 file as well, but there's a fair amount different to review.  Also the session 5 file except_exercise.py